pkg/controller/statefulset/stateful_set_control_test.go

```diff
@@ -2474,6 +2474,8 @@ func (om *fakeObjectManager) CreatePod(ctx context.Context, pod *v1.Pod) error {
 	defer om.createPodTracker.inc()
 	if om.createPodTracker.errorReady() {
 		defer om.createPodTracker.reset()
+		defer om.createPodTracker.Unlock()
```

Comment:
I've tried running this for 20mins locally, which on master was enough to trigger the race and that seemed to fix it. But then I looked at the code carefully, and I see that the `requestTracker` which here is under `createPodTracker` has its own 2 locks already. One is on parallel operation tracker, and another on the whole tracker, which helps syncing operations like `reset` here. Based on that, I'm not sure adding another layer of locks is advisable, I'd try to check why the current ones are not sufficient, before jumping on adding yet another one.

Reply:
Locks are added on ```requestTracker``` throughout whenever internal fields like ```parallel``` and ```err``` are getting accessed. However err field is getting accessed through ```om.createPodTracker``` without obtaining lock and reset function is getting called after that so I felt the need of adding it here as well.

Reply:
@srivastav-abhishek ok, I see what you're saying. In that case we have two possible outcomes here:
1. add similar lock and unlock invocations to all `fakeObjectManager` methods which are returning the err field from `requestTracker`
2. add a new method to `requestTracker` which will return err instead, and handles the lock for you. 

I'm personally leaning towards the latter, since that will centralize the locking, wdyt?

Reply:
@soltysh I agree to your suggestion of centralising locks for ```requestTracker``` to get ```err``` field. I will add changes to support this in the PR.